pkg/versionutil/version.go

```diff
@@ -33,3 +38,18 @@ func DetailedVersionString() string {
 	// e.g. v2.2.0 git:03669cef-clean build:2016-07-22T16:22:26.556103000+00:00 go:go1.6.2
 	return fmt.Sprintf("v%s git:%s-%s build:%s %s", Version, GitCommit, GitState, BuildDate, runtime.Version())
 }
+
+// Extracts the installed version of run command from the service definition.
+// Current extracting it from the ExecStart field that includes the version as a substring.
+func ExtractFromServiceDefinition(content string, ctx *log.Context) (string, error) {
+	ctx.Log("message", "extracting version from service definition "+content)
+	firstSplit := strings.Split(string(content), fmt.Sprintf("ExecStart=/var/lib/waagent/%s-", constants.RunCommandExtensionName))
```

Comment:
Is there a way to dynamically get the location of agent "/var/lib/waagent" ? just in case if the location of agent changes in future

Reply:
Not at this moment. I remember having the same conversation with Norberto and he mentioned that there are already thoughts about moving the location of the agent.

I can add a new constant for this known value to start the consolidation of this and have only one single place where it needs to be changed. I'll do it for the immediate run command changes, and we can make a separate PR for other places in code :)